.github/workflows/ci.yml

```diff
@@ -385,15 +395,87 @@ jobs:
 
       - run: ${{ matrix.ls || 'ls -lh' }} dist/
 
-      - run: twine check --strict dist/*
+      - uses: actions/upload-artifact@v3
+        with:
+          name: pypi_files
+          path: dist
+
+  build-pgo:
```

Comment:
Do we need to do two builds?

Reply:
The problem I have been encountering is that we need to run `pytest` with the correct interpreter for the build. I have started with this split approach to verify the end-to-end approach works, and then I'll attempt to merge back into a single step.

Reply:
Makes sense.